scheduler/scheduler.go

```diff
@@ -166,16 +166,27 @@ func GetDistroQueueInfo(distroID string, tasks []task.Task, maxDurationThreshold
 		if !opts.IncludesDependencies || checkDependenciesMet(&task, depCache) {
 			task.ExpectedDuration = duration
 			distroExpectedDuration += duration
-		}
-
-		if duration > maxDurationThreshold {
-			if !opts.IncludesDependencies || checkDependenciesMet(&task, depCache) {
+			startTime := task.ScheduledTime
+			if task.DependenciesMetTime.After(startTime) {
+				startTime = task.DependenciesMetTime
+			}
+			task.WaitSinceDependenciesMet = time.Now().Sub(startTime)
+			// duration is defined as expected runtime and does not include wait time
+			if duration > maxDurationThreshold {
 				if info != nil {
-					info.CountOverThreshold++
+					info.CountDurationOverThreshold++
 					info.DurationOverThreshold += duration
 				}
-				distroCountOverThreshold++
+				distroCountDurationOverThreshold++
 			}
+			// actual wait time allows us to independently check that the threshold is working
+			if task.WaitSinceDependenciesMet > maxDurationThreshold {
```

Comment:
Is it okay that tasks might be double counted? (i.e. counted for duration and wait?)

Reply:
I see what you're saying -- if we had some task that was both a Long Task and had a Long Wait, we'd count it twice. I'd argue that this is fine, as we will only double count do to long wait if we were undercounting previously by only counting it once as a Long Task, and the additional host would be useful to ensure quality of service

Reply:
That makes sense to me!